zerver/actions/message_edit.py

```diff
@@ -878,13 +878,7 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
         # request did is mark the topic as resolved.
         new_thread_notification_string = None
         if send_notification_to_new_thread and (
-            new_stream is not None
-            or not sent_resolve_topic_notification
-            or (
-                pre_truncation_topic_name is not None
-                and orig_topic_name.lstrip(RESOLVED_TOPIC_PREFIX)
-                != pre_truncation_topic_name.lstrip(RESOLVED_TOPIC_PREFIX)
-            )
+            new_stream is not None or not sent_resolve_topic_notification
```

Comment:
What happens if one changes a topic from `{RESOLVED_TOPIC_PREFIX} foo` to `bar`? I think the code here was intended to send both "topic was unresolved" and "topic was moved" notifications; I'm not entirely sure. I would have expected us to have a test that would cover this corner case.

@juliaBichler01 FYI in case you remember something here.

Reply:
Yes, I agree that this is the intended behavior.
I think the issue here is that we are missing the "topic was unresolved" notification.

Reply:
I'm not sure I understand what you mean here; can you explain?